x-pack/plugins/endpoint/common/generate_data.test.ts

```diff
@@ -93,11 +93,92 @@ describe('data generator', () => {
     expect(processEvent.process.name).not.toBeNull();
   });
 
+  describe('creates a resolver tree structure', () => {
+    let tree: Tree;
+    beforeEach(() => {
+      tree = generator.generateTree({
+        alwaysGenMaxChildrenPerNode: true,
+        ancestors: 3,
+        children: 3,
+        generations: 3,
+        percentTerminated: 100,
+        percentWithRelated: 100,
+        relatedEvents: 4,
+      });
+    });
+
+    const eventInNode = (event: Event, node: TreeNode) => {
+      const inLifecycle = node.lifecycle.find(lifecycleEvent => {
+        return lifecycleEvent === event;
+      });
+
+      const inRelated = node.relatedEvents.find(relatedEvent => {
+        return relatedEvent === event;
+      });
+
+      return (inRelated || inLifecycle) && event.process.entity_id === node.id;
+    };
+
+    it('has 2 lifecycle events for ancestors, children, and the origin', () => {
+      Array.from(tree.ancestry.values()).forEach(node => {
+        expect(node.lifecycle.length).toEqual(2);
+      });
+
+      Array.from(tree.children.values()).forEach(node => {
+        expect(node.lifecycle.length).toEqual(2);
+      });
+
+      expect(tree.origin.lifecycle.length).toEqual(2);
+    });
+
+    it('has all events in one of the tree fields', () => {
+      tree.allEvents.forEach(event => {
+        if (event.event.kind === 'alert') {
+          expect(event).toEqual(tree.alertEvent);
+        } else {
+          const ancestor = tree.ancestry.get(event.process.entity_id);
+          if (ancestor) {
+            expect(eventInNode(event, ancestor)).toBeTruthy();
+            return;
+          }
+
+          const children = tree.children.get(event.process.entity_id);
+          if (children) {
+            expect(eventInNode(event, children)).toBeTruthy();
+            return;
+          }
+
+          expect(eventInNode(event, tree.origin)).toBeTruthy();
+        }
+      });
+    });
+
+    const nodeEventCount = (node: TreeNode) => {
+      return node.lifecycle.length + node.relatedEvents.length;
+    };
+
+    it('has the correct number of total events', () => {
+      // starts at 1 because the alert is in the allEvents array
+      let total = 1;
+      Array.from(tree.ancestry.values()).forEach(node => {
+        total += nodeEventCount(node);
+      });
+
+      Array.from(tree.children.values()).forEach(node => {
+        total += nodeEventCount(node);
+      });
+
+      total += nodeEventCount(tree.origin);
+
+      expect(tree.allEvents.length).toEqual(total);
+    });
+  });
+
   describe('creates alert ancestor tree', () => {
     let events: Event[];
 
     beforeEach(() => {
-      events = generator.createAlertEventAncestry(3);
+      events = generator.createAlertEventAncestry(3, 0, 0, 0);
```

Comment:
These tests relied on their only being a `start` event for ancestors, so force the tree to be that way.